pkg/controller/controller.go

```diff
@@ -68,28 +68,48 @@ func NewManagerForRestConfig(conf *config.Config, rc *rest.Config) (ctrl.Manager
 	}
 	m.GetLogger().V(2).Info("using namespace: " + conf.NS)
 
-	if err = ingress.NewIngressControllerReconciler(m, manifests.IngressControllerResources(conf, deploy)); err != nil {
+	if err := dns.NewExternalDns(m, conf, deploy); err != nil {
 		return nil, err
 	}
-	if err = ingress.NewConcurrencyWatchdog(m, conf); err != nil {
+
+	nginxConfigs, err := nginx.New(m, conf, deploy)
+	if err != nil {
 		return nil, err
 	}
-	if err = keyvault.NewIngressSecretProviderClassReconciler(m, conf); err != nil {
+
+	watchdogTargets := make([]*ingress.WatchdogTarget, 0)
+	for _, nginxConfig := range nginxConfigs {
+		watchdogTargets = append(watchdogTargets, &ingress.WatchdogTarget{
+			ScrapeFn:    ingress.NginxScrapeFn,
```

Comment:
Might be missing something in the larger scheme of things but are we ever going to be creating watchdog targets for nginx objects that DON'T use the nginx scrape function? Why import the scrape function separately instead of just including it in the config struct? That way we can ensure unique configs can all have their own scrape function associated with them rather than making future contributors make that logical leap?

Reply:
What do you mean by import the scrape function separately? Each unique config does have their own scrape function.

Are you suggesting we should have an additional nginx config struct that automatically has that?

Reply:
Saw the later comment. I'm assuming you mean attaching it to the nginxIngressConfig struct that lives in the manifests. Keeping them separate simply allows for more flexibility in the future. Things are more flexible and less tightly coupled like this. It also gives an example of what a scrape fn would actually look like in the package it's used in which helps anyone reading that file understand.

I don't think it's a large logical leap. It's very logical that nginxConfigs would want to use the nginxScrapeFn.

Reply:
Ok, that makes sense. I see what you mean about flexibility but I personally still think that coupling the scrape function would make the separation from nginx cleaner along with making it easier to consolidate all the requirements/required functionality for if/when we add a new ingress provider. But if you feel strongly about the design here then I trust your understanding.